tests/functional/spec/components/radio.spec.js

```diff
@@ -0,0 +1,236 @@
+const helpers = require('../../helpers');
+const RadioMandatoryPage = require('../../pages/surveys/radio/radio-mandatory.page');
+const RadioNonMandatoryPage = require('../../pages/surveys/radio/radio-non-mandatory.page');
+const SummaryPage = require('../../pages/surveys/radio/radio-summary.page');
+
+describe('Radio button with an option', function() {
+
+var radio_schema = 'test_radio.json';
+
+it('Given an mandatory option is available, when the user clicks the mandatory option then they should be taken to the next page', function() {
+    return helpers.openQuestionnaire(radio_schema).then(() => {
+      return browser
+        // Nothing should be selected by default
+        .isSelected(RadioMandatoryPage.none()).should.eventually.be.false
+        .isSelected(RadioMandatoryPage.bacon()).should.eventually.be.false
+        .isSelected(RadioMandatoryPage.eggs()).should.eventually.be.false
+        .isSelected(RadioMandatoryPage.sausage()).should.eventually.be.false
+        .isSelected(RadioMandatoryPage.other()).should.eventually.be.false
+
+        // Choose Bacon option
+        .click(RadioMandatoryPage.bacon())
+        .isSelected(RadioMandatoryPage.bacon()).should.eventually.be.true
+
+        // Submit answers given on mandatory page
+        .click(RadioMandatoryPage.submit())
+
+         // Skipping non-mandatory page
+        .click(RadioNonMandatoryPage.submit())
+
+        // Then my answer is shown on the summary
+        .getUrl().should.eventually.contain(SummaryPage.pageName)
+        .getText(SummaryPage.answer_mandatory()).should.eventually.contain('Bacon');
```

Comment:
Across all of the tests we need to remove all actions and assertions not related to the test. 

For example, this test is testing choosing a mandatory option and going to the next page, so we should not be testing that nothing is selected before starting, the optional page, or the summary. 

This test should only:
- Start the survey
- Select an option
- Submit
- Assert that we're on the next page